example.py

```diff
@@ -8,36 +8,89 @@ class ExampleLoggingApp(krux.cli.Application):
     This class gets a few things for free; not least of which:
       * a command-line option parser
       * a logger
+      * a stats dispatcher
+      * a context manager, which cleans up after us,
+        including deleting the lock file, if one was created.
     """
-    def __init__(self, *args, **kwargs):
-        # the 'name' parameter is just a string/name for your script. It should be unique in your environment,
-        # because stats will be emitted to statsd with that as part of the stat name.
-
-        # the syslog_facility parameter being set changes where logs get sent to;
-        # the default is to print them out to the console; setting a syslog facility
-        # means messages will go to syslog with the named facility and called severity;
-        # your local syslog config will need to route those messages to a file or remote syslog.
-
-        # A NOTE ON *args AND **kwargs: Our intent is to capture and pass on the arguments
-        # to parent methods, which want to use them. This is one way. Another way
-        # is to declare the arguments explicitly. There's a software maintenance cost to that, though:
-        # if parameter changes are made to a ancestor method, it may require a change
-        # to the descendant methods.
-        super(ExampleLoggingApp, self).__init__(name='testloggingapp', syslog_facility='local0', *args, **kwargs)
+    def __init__(
+        self,
+        name,
+        parser=None,
+        logger=None,
+        lockfile=False,
+        syslog_facility=krux.cli.DEFAULT_LOG_FACILITY,
+        log_to_stdout=True,
+        parser_args=None,
+    ):
+        # The above method signature matches that of the parent, krux.cli.Application.__init__()
+        super(ExampleLoggingApp, self).__init__(
+            name=name,
+            parser=parser,
+            logger=logger,
+            lockfile=lockfile,
+            syslog_facility=syslog_facility,
+            log_to_stdout=log_to_stdout,
+            parser_args=parser_args,
+        )
+
+        # If you have command line arguments, this is when you want
+        # to get retrieve & store them. super().__init__() will have stored them
+        # self.args for us. The only place we want to read from self.args
+        # is here in our __init__().
```

Comment:
I love this commentary. It could be improved by explaining _why_ reading here and not elsewhere is good (testability) but that's not a blocker for this pull.